src/composables/authSignupFunction.js

```diff
@@ -0,0 +1,22 @@
+import { createUserWithEmailAndPassword, updateProfile } from "firebase/auth";
+import { appAuth } from "@/firebase/config";
+
+async function authSignUp(name, email, password) {
+  try {
+    const userCredential = await createUserWithEmailAndPassword(
+      appAuth,
+      email,
+      password
+    );
+    const user = userCredential.user;
```

Comment:
Don't you think they should be a guard here to make sure the user is found before updating user profile in line 12?

Reply:
Not completely sure what you mean here. Check to see if the potential new user already has an account in Firebase?

Reply:
`const user = userCredential.user`
what if user = undefined

` if user == undefined || user == null`
the next line
line 12 `await updateProfile(user, {displayName:name});` will lead to an error